src/main.rs

```diff
@@ -48,10 +53,21 @@ enum Commands {
 fn main() {
     let cli = Cli::parse();
 
-    let ssh_user = cli.user.unwrap_or(env::var("USER").unwrap());
+    let ssh_user = match cli.user {
+        Some(user) => user,
+        None => (|| -> Option<String> {
+            let mut reader = BufReader::new(
+                File::open(Path::new((env::var("HOME").unwrap().to_owned() + "/.ssh/config").as_str())).ok()?
```

Comment:
Personally I would try to work with Path join rather than string concat but that's not that important.
https://doc.rust-lang.org/rust-by-example/std_misc/path.html
```diff
-File::open(Path::new((env::var("HOME").unwrap().to_owned() + "/.ssh/config").as_str())).ok()?
+File::open(Path::new(env::var_os("HOME").unwrap()).join(".ssh").join("config")).ok()?
```

using var_os is even cleaner here as it allows for non Utf8 paths which are possible on the filesystem.